src/http_rest_api_handler.cpp

```diff
@@ -204,24 +204,227 @@ std::string HttpRestApiHandler::preprocessInferRequest(std::string request_body)
     return buffer.GetString();
 }
 
-::inference::ModelInferRequest HttpRestApiHandler::prepareGrpcRequest(const std::string modelName, const std::string modelVersion, const std::string request_body) {
-    ::inference::ModelInferRequest grpc_request;
+Status convertStringToVectorOfSizes(const std::string& comma_separated_numbers, std::vector<int>& sizes) {
+    std::stringstream streamData(comma_separated_numbers);
+
+    std::string numberString;
+    while (std::getline(streamData, numberString, ',')) {
+        int binarySize;
+        try {
+            binarySize = std::stoi(numberString);
+        } catch (const std::invalid_argument& ia) {
+            SPDLOG_ERROR("Invalid argument in binary size string: {}", numberString);
+            return StatusCode::REST_BINARY_DATA_SIZE_PARAMETER_INVALID;
+        } catch (const std::out_of_range& oor) {
+            SPDLOG_ERROR("Argument in binary size string out of range: {}", numberString);
+            return StatusCode::REST_BINARY_DATA_SIZE_PARAMETER_INVALID;
```

Comment:
Those messages can be triggered in predict path correct? If so then this should be DEBUG